### PR TITLE
GAEN-processes docs

### PR DESCRIPTION
**Pull request by sindremb**
Added diagrams and descriptions of the processes in the solution related to using GAEN for exposure notification + some basic high level information about the solution to the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,36 @@
-# Documentation repository for Smittestopp
+# Smittestopp (NO exposure notification app)
+
+## Components overview
+
+An overview of the different components involved in the Smittestopp application is illustraded by the diagram below.
```
Comment (matishadow) [inline README.md]: Typo in word:
should be `illustrated` not `illustraded`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,36 @@
...
+
+The [processes and data flows for exposure notification](gaen-processes.md) between the different components are covered in more detail in a [separate document](gaen-processes.md).
+
+### Smittestopp backend
+
+The central backend responsible for distributing the diagnosis keys used by the app to detect exposures.
+
+### Smittestopp app
+
+The application running on each users phone, responsible for activating the Exposure Notification framework to start the distributed contact tracing, receving diagnosis keys from the backend to check for exposure, and uploading diagnosis keys upon verified infection for the user.
```
Comment (matishadow) [inline README.md]: I think it should be `on each user's phone`

Comment (matishadow) [inline README.md]: Typo in `receving diagnosis`, should be `receiving diagnosis`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,36 @@
...
+
+### Smittestopp verification
+
+The solution responsible for verifying the infection status of a user and gratning access to notifying their contacts of exposure.
```
Comment (matishadow) [inline README.md]: Should be `granting access` not `gratning access`

File: diagrams/Smittestopp_receive_notification_en.drawio [hunk 4]
```diff
@@ -0,0 +1 @@
...wUkxho4LvHlcS7nIU0iImSc2QPKx3rQXE7lb3HuRbnzPDPCdRJPPBp4l016jxxtOhUH9Sd7Vzsld3cBr4Gpi4ed9z0zKfxenQB+fkQcbSQRChKpVe7PbNJujniENgFJLiOf42tJrWvGfVLL26p2kVdQufl+gRZob+eI8wOR3aI6aWRyzNw+XVoqnZGBoC06F6zDsYmNpVyCXHJVDc47GYLA5eWmRiEyVmMWU5KATosTSRHumLxBeDhnNSYDGDp4j3swZGqpuRuBKr5HxxutpJBQRJiCFGomblqo5jEhLQuJQgBY0cW+7RFN8F5bdZsus6qj3ijnd5g5FqV8CNFMln2Wg5muzHGcxEcDoZcIbJgDO4ZhW1v26H8z9a7wWt3cAI9EcXvgajpy58A7vw/fTj1Lnb8NtV504H3OhgdW4wEJDfXed4+DoOrwlXF/OzcfOuLm4dT7+2F7fiZdN5OceM5Fie4e3N7sAdsHfXJfCtXvGdXwo+9C7PvGsb33E3Z97lPXR84PuGcx/gLi/wLTRyIR9wLoUv9hOKMMXgg68RlMy8obHUM4KSXZGrnHApKIgPNpB+AMsEwbNb5mHl94vslVi5lpkiPTbXmptJ26FzLbsyVkXYJ8pJvOkF0SuMHjPZmg7cVW4jbN/Rk2QMv7uMnJxXi/Tmy9VZ+OFq4K8ju7vct7SsZdP58/LDSAh/qgAQh5zQuoJq+e6WtqIIBooTzkvKkFRdp3PyAW/srvxPESKycr4RXxcy2FRPX5TYMaM5/KiLEqcuZVEOBiMxUSW4lrP4+Z4yXpy/VzJeUA4hXSTweM7omuSECyneR+AegjsbEDWkBUch7wrb6xaAAnbL0WvJ602BK4c1wIvsSiBpcjAnEUxB2KpVAk+FEktGlWpgsUrJgMTVtBPSXDi37G3gSgzltOXBcEhZ1IqqmEX4KwnlQ82ULtotMpyQistrBpHAnBRCPTnikLbodWHFa1JInisC/m+7jRNDIDeLSf+SOgNZct2EwUqHxBwMG6kzbR/jYGb5VqjPcqyQGctUU6sCkKUOuZpB4y3PYT4j3ciJcAnnDJZw21EROFulRklOjSdKq8vN1LwlR1Ruq6DbVE+wAGwK074owook6tix6UelKGrV/GoaSUee378dnwz8ucEfOoG9R3SS4LX9W7o6+tq//vvL/wA=</diagram></mxfile>
```
Comment (matishadow) [inline diagrams/Smittestopp_receive_notification_en.drawio]: Typo in there. There's `tese to RPIs` fragment there. Should be `these to RPIs`

File: diagrams/Smittestopp_notify_contacts_en.drawio [hunk 5]
```diff
@@ -0,0 +1 @@
...D0yntKluttNyhRcqKAFD4vasm9MPnO4PMlWh8ch9XPZAGyoSAUpG7//SIPCNnDIy5SaxDIhwZGDvIc/tcNTj/zROCAJswRouMy1GQ4vd4uWNe+8ZqxXnz5fbnHgzsWhMiJc246t9Upq6MEexKG6j43VX6bpr6E9lJSZ95tv/CHbQnid/hvY6so6svubRZ8uHq9pNl/7YxusylQWP5pn0Yz/QwnXtyu96VHGzzZWDv4ZOvzv7Dm48xRLYO9wT0fL3+hWvmbbrW46Mq2sDejG/Z/6UrWjPQ23Uv5YUljY+8mTLY7m3jj0TspozYNQ9rM+LIidjAThmcuQs19851CeeuT24O50HtnMAdwzmIUbm8IfjOwhFieohSsEo2pbkwh4NEOodkCQwxx8WcFKkIhES21lS1ToS6VbfZrNKmfMzwPKY8XVu1Fqd17Vaqt6k7L0gqSIZzMTz7U97qq62MsFX80AkpP7Fzg3UcM9Qp5a3F2eKL+FjvYrivVdrg2LBrle7yTrU32D/J7FwGvG2o9ycCTBIP7TMnRwRZ9QsJzi2gUiQSiN27ixRX5T0Pq74B5kNwUowyuXdNMFXWoCiF/+QnfHosjwWJI+4NFpHxjIyIbmEW8NuhHl1yTqWyG14kVwVhoggJkYh6EIzO51iAzNGurNAHBRmJUcuhU0Fj0cU9FjxlBRLbtEjz1FmZTkXHomxVIpwaE1S+3uF/Kcn6q0PhpkS8HvrSsmzMWpalYvqQQlnLfrZ1z9NxTTcNP34TCSHw1qACV3RgRC8AA/N1j/Y/WojqzeCnGcYhFXLB6nYiwFwKAGgpHkw6MbkrmFpYym50pzNPENCKYlZBJ185QdPchcyAQEJ3NFaqv2arJgHifeZM8jPDjz0reQHkEHkN2t8QvHeCt4Igdb/I9M+ieaRPOlFlgTNcIKbxpj23JvXMs2hLIeCWU4o24ytwSeXooDM4TDpmLCdSmdRWLFfncapm0xJ+QEu+NcQP8UwQ8Y2YMnJRglan0thvN7UbGE5+hFpBX2BWFtCUsZsW0gsCat2nyBnWU2sf/xuqUeeqJj+c/PaVuxT4jMBhWLD5MWaRBpTiOHxF5oRoSXWTG91ICOHuBKtaXHU7S4wEACQJ++ey0I1Go/Y6N7DXOb/6BQR9nfPNuGeDdY6fNr9bJuVv89tw/u3/AQ==</diagram></mxfile>
```
Comment (matishadow) [inline diagrams/Smittestopp_notify_contacts_en.drawio]: `Status: Positiv` should be changed to `Status: Positive`

Comment (matishadow) [inline diagrams/Smittestopp_notify_contacts_en.drawio]: `Completed authenticatino` to `Completed authentication`

File: gaen-processes.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# Exposire Notification Processes
```
Comment (matishadow) [inline gaen-processes.md]: `Exposure`

File: gaen-processes.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+This document describes the processes, related to GAEN, used in the app to perform distributed contact registration, letting an infected user notify their digital contacts, and notifying app users of exposure.
```
Comment (matishadow) [inline gaen-processes.md:3]: `registration` not `registration`

File: gaen-processes.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+## Distributed contact registration
+
+The figure below illustrates how the distributed contact registation works between two different users of an app using the GAEN-framework.
+
+Once actived through consent from the user, the apps will generate daily keys called Temporary Exposure Keys (TEKs). From these keys, a Rotating Proximity Identifier (RPI) is generated every 5-10 minutes. The RPI is then broadcasted through Bluetooth Low Energy to all other devices listening for the same service ID. The broadcasted RPIs are detected by other devices through frequent scanning, and is recorded on the device, along with timestamp and signal strength. This information can later be used to detect exposure from any TEKs linked to infection shared with the device.
```
Comment (matishadow) [inline gaen-processes.md]: `Once activated`

Comment (matishadow) [inline gaen-processes.md]: According to markdown best practices, this paragraph `Once actived through consent from the user,...` should be split.
One sentence per one line.

File: gaen-processes.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+![Smittestopp components overview](diagrams/Smittestopp_contact_registration_en.png)
+## Verify infection and notify exposure
+
+The figure below illustrates how the a user can notify their digital close contacts through the application.
```
Comment (matishadow) [inline gaen-processes.md]: Please change `the a user` to `a user` (or `the user if you'd like).

File: gaen-processes.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+Once a user has been informed that they have tested positive for COVID-19, they may start the notification process through the app. The first step of this process is to verify the infection status of the user. To start this process, the user has to give an additional consent to the app. Given consent, the app opens up the verification solution, where the user will be asked to authenticate through ID-porten. After succsessful authentication, the verification solution performs a lookup in MSIS on behalf of the authenticated user, using the national identifier provided by ID-porten, to verify that there exists a valid positive test result. To successfully verify infection status, the user must be at least 16 years of age, and have a positive COVID-19 test result with the sample taken within the last 14 days.
```
Comment (matishadow) [inline gaen-processes.md]: Please change `succsessful` to `successful`

Comment (matishadow) [inline gaen-processes.md]: According to markdown best practices, only one sentence should be in one line.

File: gaen-processes.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+The verification solution keeps track of how many verifications have been performed by the same user within the last 24 hours. If a user has exceeded 3 verifications in the relevant time period, they will be flagged as blocked from performing the actual notification process. The performed verifications are recorded using a hash of the pseudonym provided from ID-porten.
```
Comment (matishadow) [inline gaen-processes.md]: According to markdown best practices, only one sentence should be in one line.

File: gaen-processes.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+Once the verification is completed, the user is returned to the app with an access token containing both the information neede by the app to continue the process, as well as the necessary claims to perform an upload of diagnosis keys to the central backend.
+
+If the verification either does not confirm infection status, or has flagged the user as blocked, the app will display a relevant error message and end the process. Otherwise, the app will ask the user for some additional information to continue the process. The user is asked to provide the date of onset of symptoms, which is used to assign infectiousness scores to each TEK to be uploaded. If no date of symptoms is used, the sample date is used instead.
```
Comment (matishadow) [inline gaen-processes.md]: According to markdown best practices, only one sentence should be in one line.

File: gaen-processes.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+## Detecting exposure
+
+The figure below illustrates the process for detecting and notifying a user of exposure.
+
+Through the GAEN-framework, the app makes sure that en exposure check is performed periodically (typically a few times a day).
+
+The first step is to retreive new diagnosis keys (TEKs + infectiousness) from the central backend solution. These new keys are then provided to the GAEN-framework, and together with any diagnosis keys provided earlier, exposure information is proveded to the app. The way GAEN determines exposure is by calculating all valid RPIs for all TEKs provided, and then looking for matches with RPIs recorded on the device during contact registation. For any matches, the time periode a contact was recorded for, the estimated distance to the device emitting the RPI (based on signal strength), and the infectiousness value for the relevant TEK is used to determinde the risk value. In addition to risk values, the exposure information from GAEN includes rough information about duration, time of contact, etc., for qualified exposures, according to the configuration provided by the app.
```
Comment (matishadow) [inline gaen-processes.md]: According to markdown best practices, only one sentence should be in one line.

File: gaen-processes.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+The verification solution keeps track of how many verifications have been performed by the same user within the last 24 hours. If a user has exceeded 3 verifications in the relevant time period, they will be flagged as blocked from performing the actual notification process. The performed verifications are recorded using a hash of the pseudonym provided from ID-porten.
+
+Once the verification is completed, the user is returned to the app with an access token containing both the information neede by the app to continue the process, as well as the necessary claims to perform an upload of diagnosis keys to the central backend.
```
Comment (matishadow) [inline gaen-processes.md]: Please change `neede` to `needed`.

File: gaen-processes.md (hunk 13 above)
Comment (matishadow) [inline gaen-processes.md]: `retrieve` not `retreive`

Comment (matishadow) [inline gaen-processes.md]: `period` not `periode`

Comment (matishadow) [inline gaen-processes.md]: `determine` not `determinde`

**Review by matishadow — CHANGES_REQUESTED** [review]

File: gaen-processes.md (hunk 7 above)
Comment (sindremb) [reply gaen-processes.md:3]: Correction an actual content the same here?

**Review by matishadow — APPROVED** [review]

**Comment by sindremb** [comment]
Thanks a lot for a thorough review, @matishadow

**Pull request merged**